### PR TITLE
Support SNS FIFO queues

### LINKED ISSUE #715: [BUG] SNS Fifo queues are not supported
**What is the bug?**
SNS FIFO queues are not supported.

**How can one reproduce the bug?**
Send a message to a SNS FIFO queue

### PR DESCRIPTION
**Pull request by lezzago**
### Description
Support SNS FIFO queues

### Issues Resolved
#715 

### Check List
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by lezzago** [comment]
Failing tests are due to BWC failure that have been existing.

File: notifications/core/src/main/kotlin/org/opensearch/notifications/core/setting/PluginSettings.kt [hunk 1]
```diff
@@ -464,6 +483,11 @@ internal object PluginSettings {
             log.debug("$LOG_PREFIX:$HOST_DENY_LIST_KEY -autoUpdatedTo-> $clusterHostDenyList")
             hostDenyList = clusterHostDenyList
         }
+        val clusterClusterName = clusterService.clusterSettings.get(CLUSTER_NAME_SETTING)
+        if (clusterClusterName != null) {
+            log.debug("$LOG_PREFIX:$CLUSTER_NAME_SETTING -autoUpdatedTo-> $clusterClusterName")
+            clusterName = clusterClusterName.value()
+        }
     }
```
Comment (Hailong-am) [inline notifications/core/src/main/kotlin/org/opensearch/notifications/core/setting/PluginSettings.kt:491]: how about use `clusterService.clusterName.value()`?

File: notifications/core/src/main/kotlin/org/opensearch/notifications/core/setting/PluginSettings.kt [hunk 2]
```diff
@@ -411,6 +429,7 @@ internal object PluginSettings {
         tooltipSupport = TOOLTIP_SUPPORT.get(clusterService.settings)
         hostDenyList = HOST_DENY_LIST.get(clusterService.settings)
         destinationSettings = loadDestinationSettings(clusterService.settings)
+        clusterName = clusterService.settings.get(CLUSTER_NAME)
```
Comment (Hailong-am) [inline notifications/core/src/main/kotlin/org/opensearch/notifications/core/setting/PluginSettings.kt:432]: same here, how about use `clusterService.clusterName.value()` directly?

File: notifications/core/src/main/kotlin/org/opensearch/notifications/core/setting/PluginSettings.kt (hunk 1 above)
Comment (lezzago) [reply notifications/core/src/main/kotlin/org/opensearch/notifications/core/setting/PluginSettings.kt:491]: Updated

File: notifications/core/src/main/kotlin/org/opensearch/notifications/core/setting/PluginSettings.kt (hunk 2 above)
Comment (lezzago) [reply notifications/core/src/main/kotlin/org/opensearch/notifications/core/setting/PluginSettings.kt]: updated

**Review by Hailong-am — DISMISSED** [review]

**Review by Hailong-am — COMMENTED** [review]
have few unit test failed, need fix failed unit test first.

**Review by Hailong-am — APPROVED** [review]

**Review by gaobinlong — APPROVED** [review]

**Pull request merged**